### PR TITLE
[MAINT] add test for second level input as list of 3D image filenames

### LINKED ISSUE #3642: add test for check_second_level_input to accept 3D images
@Remi-Gau I will merge this to green the CI and then I agree to do a follow-up PR to add a test.

_Originally posted by @ymzayek in https://github.com/nilearn/nilearn/issues/3637#issuecomment-1493299436_

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3642
Act as regression test for #3636 

Changes proposed in this pull request:

- add a test that pass but would fail if #3637 is reverted
- this checks that a second level model fit can accept a list of 3D image filenames as input

### CODE REVIEW

File: nilearn/glm/tests/test_second_level.py [hunk 1]
```diff
@@ -98,6 +99,37 @@ def test_sort_input_dataframe(input_df):
...
     )
 
+def test_second_level_input_as_3D_images():
+    """Test second level model with a list 3D image filenames as input.
+
+    Should act as a regression test for:
+    https://github.com/nilearn/nilearn/issues/3636
+
+    Adapted from examples: plot_second_level_one_sample_test.py
+
```
Comment (Remi-Gau) [inline nilearn/glm/tests/test_second_level.py:109]:
```suggestion
```

**Review by bthirion — APPROVED** [review]
LGTM.

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @Remi-Gau !

**Comment by Remi-Gau** [comment]
Had to lint.
Will merge after CI finishes.

**Pull request merged**